<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QueryBuilderWizardForm</class>
 <widget class="QWizard" name="QueryBuilderWizardForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>640</width>
    <height>480</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>640</width>
    <height>480</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Wizard</string>
  </property>
  <property name="wizardStyle">
   <enum>QWizard::ModernStyle</enum>
  </property>
  <property name="options">
   <set>QWizard::HaveHelpButton|QWizard::IndependentPages</set>
  </property>
  <widget class="QWizardPage" name="m_dataSourceWizardPage">
   <property name="title">
    <string>Query Builder</string>
   </property>
   <property name="subTitle">
    <string>Creates a query</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Data Source</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_20">
         <item row="0" column="0">
          <widget class="QComboBox" name="m_dataSourceComboBox"/>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>SQL Dialect</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_21">
           <item row="0" column="0">
            <widget class="QTextEdit" name="m_dsSQLDialectTextEdit">
             <property name="documentTitle">
              <string/>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt; font-weight:600;&quot;&gt;SQL query dialect accepted &lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;justify&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:8pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="m_dataSetWizardPage">
   <property name="title">
    <string>Data Sets (FROM)</string>
   </property>
   <property name="subTitle">
    <string>Select data sets</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_11">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_10">
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_6">
        <property name="title">
         <string>Data Set</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_8">
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="0" column="0">
              <layout class="QGridLayout" name="gridLayout_4">
               <item row="0" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>Name</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Alias</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="0" column="1">
              <layout class="QGridLayout" name="gridLayout_5">
               <item row="0" column="0">
                <widget class="QComboBox" name="m_dataSetComboBox"/>
               </item>
               <item row="1" column="0">
                <widget class="QLineEdit" name="m_dataSetAliasLineEdit">
                 <property name="readOnly">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item row="0" column="1">
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Minimum</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="2">
            <layout class="QGridLayout" name="gridLayout_30">
             <item row="0" column="0">
              <widget class="QPushButton" name="m_addDataSetPushButton">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="m_removeDataSetPushButton">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QGroupBox" name="groupBox_7">
        <property name="title">
         <string/>
        </property>
        <layout class="QGridLayout" name="gridLayout_9">
         <item row="0" column="0">
          <widget class="QTableWidget" name="m_dataSetTableWidget">
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <attribute name="horizontalHeaderMinimumSectionSize">
            <number>29</number>
           </attribute>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Data Set</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Alias</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="m_propertyWizardPage">
   <property name="title">
    <string>Properties (FIELDS)</string>
   </property>
   <property name="subTitle">
    <string>Select attributes</string>
   </property>
   <layout class="QGridLayout" name="m_propertyGridLayout"/>
  </widget>
  <widget class="QWizardPage" name="m_whereClauseWizardPage">
   <property name="title">
    <string>Where Clause</string>
   </property>
   <property name="subTitle">
    <string>Defines a restriction clause.</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_36">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_35">
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>Criteria</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_34">
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_33">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_32">
             <item row="0" column="0">
              <layout class="QGridLayout" name="gridLayout_17">
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="minimumSize">
                  <size>
                   <width>100</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>100</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Restrict Value</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="m_restrictValueComboBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="1" column="0">
              <layout class="QGridLayout" name="gridLayout_19">
               <item row="0" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="minimumSize">
                  <size>
                   <width>100</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>100</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Operator</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="m_OperatorComboBox"/>
               </item>
              </layout>
             </item>
             <item row="2" column="0">
              <layout class="QGridLayout" name="gridLayout_12">
               <item row="0" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="minimumSize">
                  <size>
                   <width>100</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>100</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Value</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <layout class="QGridLayout" name="gridLayout_16">
                 <item row="0" column="0">
                  <layout class="QGridLayout" name="gridLayout_13">
                   <item row="0" column="0">
                    <widget class="QRadioButton" name="m_valuePropertyRadioButton">
                     <property name="text">
                      <string>Property</string>
                     </property>
                    </widget>
                   </item>
                   <item row="1" column="0">
                    <widget class="QRadioButton" name="m_valueValueRadioButton">
                     <property name="text">
                      <string>Value</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="0" column="1">
                  <layout class="QGridLayout" name="gridLayout_15">
                   <item row="1" column="0">
                    <widget class="QComboBox" name="m_valueValueComboBox">
                     <property name="enabled">
                      <bool>false</bool>
                     </property>
                     <property name="minimumSize">
                      <size>
                       <width>200</width>
                       <height>0</height>
                      </size>
                     </property>
                     <property name="editable">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                   <item row="0" column="0">
                    <widget class="QComboBox" name="m_valuePropertyComboBox">
                     <property name="enabled">
                      <bool>false</bool>
                     </property>
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                       <horstretch>0</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                     <property name="minimumSize">
                      <size>
                       <width>200</width>
                       <height>0</height>
                      </size>
                     </property>
                     <property name="editable">
                      <bool>false</bool>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item row="3" column="0">
              <layout class="QGridLayout" name="gridLayout_22">
               <item row="0" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="minimumSize">
                  <size>
                   <width>100</width>
                   <height>0</height>
                  </size>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>100</width>
                   <height>16777215</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>Connector</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="m_connectorComboBox"/>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item row="0" column="1">
            <layout class="QGridLayout" name="gridLayout_14">
             <item row="0" column="0">
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="m_addWhereClausePushButton">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <spacer name="verticalSpacer_3">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>17</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="3" column="0">
              <widget class="QPushButton" name="m_removeWhereClausePushButton">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <spacer name="verticalSpacer_2">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string/>
        </property>
        <layout class="QGridLayout" name="gridLayout_18">
         <item row="0" column="0">
          <widget class="QTableWidget" name="m_whereClauseTableWidget">
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <attribute name="horizontalHeaderMinimumSectionSize">
            <number>29</number>
           </attribute>
           <column>
            <property name="text">
             <string>Property</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Operator</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Value</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Connector</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="m_groupByWizardPage">
   <property name="title">
    <string>Group by</string>
   </property>
   <property name="subTitle">
    <string>Creates a grouping.</string>
   </property>
   <layout class="QGridLayout" name="m_groupByGridLayout"/>
  </widget>
  <widget class="QWizardPage" name="m_orderByWizardPage">
   <property name="title">
    <string>Order By</string>
   </property>
   <property name="subTitle">
    <string>Ordering the query result.</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_31">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_29">
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_14">
        <property name="title">
         <string>Order</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_27">
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_26">
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_25">
             <item row="0" column="0">
              <layout class="QGridLayout" name="gridLayout_23">
               <item row="0" column="0">
                <widget class="QLabel" name="label_12">
                 <property name="text">
                  <string>Property</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_13">
                 <property name="text">
                  <string>Order</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="0" column="1">
              <layout class="QGridLayout" name="gridLayout_24">
               <item row="0" column="0">
                <widget class="QComboBox" name="m_orderPropertyComboBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QComboBox" name="m_orderComboBox">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>100</width>
                   <height>16777215</height>
                  </size>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item row="0" column="1">
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Minimum</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="2">
            <layout class="QGridLayout" name="gridLayout_55">
             <item row="0" column="0">
              <widget class="QPushButton" name="m_addOrderPushButton">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QPushButton" name="m_removeOrderPushButton">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QGroupBox" name="groupBox_15">
        <property name="title">
         <string/>
        </property>
        <layout class="QGridLayout" name="gridLayout_28">
         <item row="0" column="0">
          <widget class="QTableWidget" name="m_orderTableWidget">
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <attribute name="horizontalHeaderMinimumSectionSize">
            <number>29</number>
           </attribute>
           <column>
            <property name="text">
             <string>Property</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Order</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>m_dataSourceComboBox</tabstop>
  <tabstop>m_dsSQLDialectTextEdit</tabstop>
  <tabstop>m_dataSetComboBox</tabstop>
  <tabstop>m_dataSetAliasLineEdit</tabstop>
  <tabstop>m_addDataSetPushButton</tabstop>
  <tabstop>m_removeDataSetPushButton</tabstop>
  <tabstop>m_dataSetTableWidget</tabstop>
  <tabstop>m_restrictValueComboBox</tabstop>
  <tabstop>m_valuePropertyComboBox</tabstop>
  <tabstop>m_addWhereClausePushButton</tabstop>
  <tabstop>m_removeWhereClausePushButton</tabstop>
  <tabstop>m_whereClauseTableWidget</tabstop>
  <tabstop>m_orderPropertyComboBox</tabstop>
  <tabstop>m_orderComboBox</tabstop>
  <tabstop>m_addOrderPushButton</tabstop>
  <tabstop>m_removeOrderPushButton</tabstop>
  <tabstop>m_orderTableWidget</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>m_valuePropertyRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_valuePropertyComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_valueValueRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_valueValueComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
